crates/bevy_ecs/src/system/query.rs

```diff
@@ -11,6 +11,79 @@ use std::{any::TypeId, fmt::Debug};
 use thiserror::Error;
 
 /// Provides scoped access to a [`World`] according to a given [`WorldQuery`] and query filter.
+///
+/// Queries are a powerful tool enabling the programmer to iterate over entities and their components
+/// as well as filtering them on certain conditions.
+///
+/// # Query Building Primer
+///
+/// ### Basic Component Access
+///
+/// A basic query looks like `Query<&MyComponent>` and all it does is grant immutable access to all
+/// `MyComponent` components. Similarly using `&mut MyComponent` instead grants mutable access instead.
+///
+/// The main way to access the components of a query is through the [`Query::iter`] and [`Query::iter_mut`]
+/// functions which return a [`QueryIter`] to iterate over:
+///
+/// ```
+/// # use bevy_ecs::system::Query;
+/// struct MyComponent(usize);
+/// fn system(query: Query<&MyComponent>) {
+///     for MyComponent(number) in query.iter() {
```

Comment:
`MyComponent(number)` is very atypical syntax here. Does this destructure it to grab the internal usize?

Reply:
Yeah, that's destructing. If you're going to use this, I'd change the struct name to be more semantically significant, e.g. `UnitHealth`, then change the binding identifier to `health`.

Also, I'd make the field of the type `pub` (even though every possible accessor already has private access).
I.e. 
```rust
struct UnitHealth(pub u32);
```
This is because it makes it clearer that the type is 'plain old data'

Reply:
That's a good point for clarity yes 👍